localstack/services/sns/provider.py

```diff
@@ -109,173 +96,32 @@ def on_after_init(self):
-        return moto_response
-
-    def list_endpoints_by_platform_application(
-        self, context: RequestContext, platform_application_arn: String, next_token: String = None
-    ) -> ListEndpointsByPlatformApplicationResponse:
-        moto_response: ListEndpointsByPlatformApplicationResponse = call_moto(context)
-        return moto_response
-
-    def list_origination_numbers(
-        self,
-        context: RequestContext,
-        next_token: nextToken = None,
-        max_results: MaxItemsListOriginationNumbers = None,
-    ) -> ListOriginationNumbersResult:
-        moto_response: ListOriginationNumbersResult = call_moto(context)
-        return moto_response
-
-    def list_phone_numbers_opted_out(
-        self, context: RequestContext, next_token: String = None
-    ) -> ListPhoneNumbersOptedOutResponse:
-        moto_response: ListPhoneNumbersOptedOutResponse = call_moto(context)
-        return moto_response
-
-    def list_platform_applications(
-        self, context: RequestContext, next_token: String = None
-    ) -> ListPlatformApplicationsResponse:
-        moto_response: ListPlatformApplicationsResponse = call_moto(context)
-        return moto_response
-
-    def list_sms_sandbox_phone_numbers(
-        self, context: RequestContext, next_token: nextToken = None, max_results: MaxItems = None
-    ) -> ListSMSSandboxPhoneNumbersResult:
-        moto_response: ListSMSSandboxPhoneNumbersResult = call_moto(context)
-        return moto_response
-
-    def list_subscriptions_by_topic(
-        self, context: RequestContext, topic_arn: topicARN, next_token: nextToken = None
-    ) -> ListSubscriptionsByTopicResponse:
-        moto_response: ListSubscriptionsByTopicResponse = call_moto(context)
-        return moto_response
-
-    def list_topics(
-        self, context: RequestContext, next_token: nextToken = None
-    ) -> ListTopicsResponse:
-        moto_response: ListTopicsResponse = call_moto(context)
-        return moto_response
-
-    def opt_in_phone_number(
-        self, context: RequestContext, phone_number: PhoneNumber
-    ) -> OptInPhoneNumberResponse:
-        call_moto(context)
-        return OptInPhoneNumberResponse()
-
-    def remove_permission(
-        self, context: RequestContext, topic_arn: topicARN, label: String
-    ) -> None:
-        call_moto(context)
-
-    def set_endpoint_attributes(
-        self, context: RequestContext, endpoint_arn: String, attributes: MapStringToString
-    ) -> None:
-        call_moto(context)
-
-    def set_platform_application_attributes(
-        self,
-        context: RequestContext,
-        platform_application_arn: String,
-        attributes: MapStringToString,
-    ) -> None:
-        call_moto(context)
-
-    def set_sms_attributes(
-        self, context: RequestContext, attributes: MapStringToString
-    ) -> SetSMSAttributesResponse:
-        call_moto(context)
-        return SetSMSAttributesResponse()
```

Comment:
These definitions are redundant - the provider will fall back to Moto for unimplemented methods